incubator/virtualcluster/pkg/syncer/resources/secret/dws.go

```diff
@@ -69,45 +69,83 @@ func (c *controller) Reconcile(request reconciler.Request) (reconciler.Result, e
 func (c *controller) reconcileSecretCreate(cluster, namespace, name string, secret *v1.Secret) error {
 	switch secret.Type {
 	case v1.SecretTypeServiceAccountToken:
-		return c.reconcileServiceAccountSecretUpdate(cluster, namespace, name, secret)
+		return c.reconcileServiceAccountSecretCreate(cluster, namespace, name, secret)
 	default:
 		return c.reconcileNormalSecretCreate(cluster, namespace, name, secret)
 	}
 }
 
-func (c *controller) reconcileServiceAccountSecretUpdate(cluster, namespace, name string, secret *v1.Secret) error {
+func (c *controller) reconcileServiceAccountSecretCreate(cluster, namespace, name string, vSecret *v1.Secret) error {
 	targetNamespace := conversion.ToSuperMasterNamespace(cluster, namespace)
+	saName := vSecret.GetAnnotations()[v1.ServiceAccountNameKey]
 
-	saName := secret.Annotations[v1.ServiceAccountNameKey]
+	c.serviceAccountTokenMutex.Lock()
```

Comment:
Here is my suggestion: add a label to record the vSecret name in the pSecret Labels. Then you can use three labels: saName, namespace, vSecret name, to uniquely identify the new pSecret for the corresponding vSecret in the super master. This can remove the need of the lock and simplify the logic of serviceAccountSecretUpdate.

Reply:
pSecret cannot be bind to vSecret. Because vSecret can be deleted by tenant, token controller would create a new one according to vSA.
mutex here to avoid two go routine create the pSecret using the generate name